### ISSUE TITLE
Request: KaGuYa engine (message.dat)

### ISSUE THREAD

**Issue by Sisulizer** [arcusmaximus/VNTranslationTools#56]
For reference: https://github.com/AyamiKaze/Galgame_Project/blob/master/KAGUYA/extKAGUYA_SCR.cpp
https://github.com/crskycode/Kaguya_Tool/blob/main/MsgTool/MsgFile.cs
Although the second tool can create new message.dat, but I still prefer vntextpatch for word wrapping (line breaker is **\n**)
[message.zip](https://github.com/arcusmaximus/VNTranslationTools/files/8946798/message.zip)

**Comment by arcusmaximus** [comment]
The first tool is for an older version of the format, but the second one works for your sample file. Even then, however, it looks like this format has the character names separated from the messages - so that if you just extract the lists as-is like that second tool does, it's not clear which names belong to which messages. If you're looking at a dialogue sheet and you don't know who's saying what, it becomes pretty hard to translate. ~~Unless you're one of those irresponsible machine translators...~~

In short, I'd need to investigate this engine by myself to find out more about the format, so that I can link messages to their names. Which game does this message.dat come from?

**Comment by Sisulizer** [comment]
The message.dat is from Mama x holic, Atelier Kaguya Bare&Bunny and Atelier Kaguya HonkyTonk Pumpkin uses same engine that has message.dat(ver 4.0).
I'm actually helping my friend with hacking the game, and manual wrapping is kinda pain. Can you also check the d3d9, the game reference it but crashed after input protag's name

**Comment by arcusmaximus** [comment]
Kaguya support is now available. I didn't get the d3d9 crash with the current build.
